src/annmas/annotate/command.py

```diff
@@ -125,36 +137,38 @@ def main(model, threads, output_bam, input_bam):
         out_header = pysam.AlignmentHeader.from_dict(out_bam_header_dict)
 
         # Start output worker:
-        output_worker = mp.Process(
-            target=_sub_process_write_fn,
-            args=(results, out_header, output_bam, pbar)
-        )
+        output_worker = mp.Process(target=_write_thread_fn, args=(results, out_header, output_bam, pbar))
         output_worker.start()
 
         # Add in a `None` sentinel value at the end of the queue - one for each subprocess - so we guarantee
         # that all subprocesses will exit:
-        iter_data = itertools.chain(bam_file, (None,) * threads)
+        iter_data = itertools.chain(bam_file, (__SENTINEL_VALUE,) * threads)
         for r in iter_data:
-            if r is not None:
-                process_input_data_queue.put(
-                    (r.to_string(), m)
-                )
-            else:
-                process_input_data_queue.put(r)
+            # We have to adjust for our sentinel value if we've got to it:
+            if r is not __SENTINEL_VALUE:
+                r = r.to_string()
+            input_data_queue.put(r)
+
+        logger.info("Finished reading data and sending it to sub-processes.")
+        logger.info("Waiting for sub-processes to finish...")
 
         # Wait for our input jobs to finish:
-        for p in worker_process_pool:
+        for p in worker_pool:
             p.join()
 
+        logger.debug("All workers stopped.")
+        logger.debug("Terminating output process.")
+
         # Now that our input processes are done, we can add our exit sentinel onto the output queue and
         # wait for that process to end:
-        results.put(None)
+        results.put(__SENTINEL_VALUE)
         output_worker.join()
 
     logger.info(f"annmas: {mod_name} finished.")
+    logger.info(f"annmas: elapsed time: %2.5fs.", time.time() - t_start)
```

Comment:
Wow!  This is *incredibly* fast now!

$ python -m annmas annotate -t 12 -o test_output.bam test_input.bam | tail -3
[INFO 2021-02-04 22:32:47] annmas annotate: annotated 100 reads with 7203 total sections.
[INFO 2021-02-04 22:32:47] annmas: annotate finished.
[INFO 2021-02-04 22:29:19] annmas: elapsed time: 14.04565s.